src/main.rs

```diff
@@ -53,7 +52,7 @@ fn write_to_handlebars<T: CodeGen>(code_gen: &T, template_path: PathBuf, output_
     file.write_all(output.as_bytes()).expect("write failed");
 }
 
-fn main() {
+fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Why is it better to have main return a result?  I thought traditionally in rust main doesn't return anything.

Reply:
It allows for the usage of `?`.